core/trino-main/src/main/java/io/trino/sql/planner/TranslationMap.java

```diff
@@ -310,6 +314,28 @@ public Expression rewriteDereferenceExpression(DereferenceExpression node, Void
                                 new LongLiteral(Long.toString(index + 1))));
             }
 
+            @Override
+            public Expression rewriteTrim(Trim node, Void context, ExpressionTreeRewriter<Void> treeRewriter)
+            {
+                Optional<Expression> mapped = tryGetMapping(node);
+                if (mapped.isPresent()) {
+                    return coerceIfNecessary(node, mapped.get());
+                }
+
+                ImmutableList.Builder<Expression> expressions = ImmutableList.builder();
+                expressions.add(node.getTrimSource());
+                node.getTrimCharacter().ifPresent(trimCharacter -> expressions.add(new Cast(trimCharacter, toSqlType(CODE_POINTS))));
+
+                ResolvedFunction resolvedFunction = analysis.getResolvedFunction(node);
+                checkArgument(resolvedFunction != null, "Function has not been analyzed: %s", node);
+
+                FunctionCall functionCall = new FunctionCall(
+                        node.getLocation().orElseThrow(),
+                        resolvedFunction.toQualifiedName(),
+                        expressions.build());
+                return coerceIfNecessary(functionCall, treeRewriter.defaultRewrite(functionCall, context));
```

Comment:
I don't quite like the idea of creating a FunctionCall with "raw" arguments, and then calling `defaultRewrite` on it.
Please begin with default-rewriting the `Trim` node, and then create the `FunctionCall`.
(this way, the `Trim` constructor without location will be needed)

Reply:
Addressed the comments except for calling `Trim` constructor without location in `TranslationMap`. Could you share the condition when doing that?